core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -435,9 +435,9 @@ protected final void doXContentAnalyzers(XContentBuilder builder, boolean includ
             boolean hasDifferentSearchQuoteAnalyzer = fieldType().searchAnalyzer().name().equals(fieldType().searchQuoteAnalyzer().name()) == false;
             if (includeDefaults || hasDefaultIndexAnalyzer == false || hasDifferentSearchAnalyzer || hasDifferentSearchQuoteAnalyzer) {
                 builder.field("analyzer", fieldType().indexAnalyzer().name());
-                if (hasDifferentSearchAnalyzer || hasDifferentSearchQuoteAnalyzer) {
+                if (includeDefaults || hasDifferentSearchAnalyzer || hasDifferentSearchQuoteAnalyzer) {
```

Comment:
this is just consistency - I didnt' have to fix this part since it was a bug in the AnalysisSerivice after all